src/components/attendee/NavOverlay/ScheduleOverlay/ScheduleOverlay.tsx

```diff
@@ -0,0 +1,196 @@
+        });
+      }
+    };
+
+    if (dayDifference <= 0) return ALWAYS_EMPTY_ARRAY;
+
+    return range(dayDifference)
+      .map((dayIndex) => {
+        const day = addDays(firstScheduleDate, dayIndex);
+
+        const daysEvents = liveAndFutureEvents.filter(
+          isEventWithinDateAndNotFinished(day)
+        );
+
+        const eventsFilledWithPriority = daysEvents.sort(
+          eventTimeAndOrderComparator
+        );
+
+        const personalisedSchedule = eventsFilledWithPriority.filter(
+          (event) => event.isSaved
+        );
+        const formattedDay = formatDayLabel(day);
+
+        return {
+          scheduleDate: formattedDay,
+          daysEvents: showPersonalisedSchedule
+            ? personalisedSchedule
+            : eventsFilledWithPriority,
+        };
+      })
+      .filter((scheduleDay) => !!scheduleDay.daysEvents.length);
+  }, [
+    isScheduleTimeshifted,
+    dayDifference,
+    liveAndFutureEvents,
+    firstScheduleDate,
+    showPersonalisedSchedule,
+  ]);
+
+  const bookmarkEvent = useCallback(
+    (
+      e: React.MouseEvent<HTMLDivElement, MouseEvent>,
+      event: ScheduledEvent
+    ) => {
+      if (!userWithId?.id) return;
+
+      // @debt get rid of stopPropagation() in the project allowing a valid event bubbling
+      e.stopPropagation();
+
+      event.isSaved
+        ? removeEventFromPersonalizedSchedule({ event, userId: userWithId.id })
+        : addEventToPersonalizedSchedule({ event, userId: userWithId.id });
+    },
+    [userWithId?.id]
+  );
+
+  const renderedEvents = useMemo(
+    () =>
+      weekdays.map(({ scheduleDate, daysEvents }) => (
+        <div key={scheduleDate}>
+          <div>{scheduleDate}</div>
+          {daysEvents.map((event) => {
+            const isCurrentEventLive = isEventLive(event);
+            const showDate = Boolean(
+              differenceInCalendarDays(
+                eventEndTime({ event }),
+                eventStartTime({ event })
+              )
+            );
+
+            return (
+              <div key={event.id}>
+                <div>{event.name}</div>
+                <div>
+                  <span>
+                    {isCurrentEventLive
+                      ? "Live "
+                      : formatTimeLocalised(eventStartTime({ event })) + " - "}
+                  </span>
+                  <span>{isCurrentEventLive && "until "}</span>
+                  <span>
+                    {showDate &&
+                      formatDateRelativeToNow(eventEndTime({ event }))}
+                  </span>{" "}
+                  <span>{formatTimeLocalised(eventEndTime({ event }))}</span>
+                  <span> in {event.spaceId}</span>
+                </div>
+                <div>{event.description}</div>
+                <div
+                  className={styles.ScheduleOverlay__bookmark}
+                  onClick={(e) => bookmarkEvent(e, event)}
```

Comment:
Should this handler be put into `useCallback` before the `useMemo` ?

Reply:
the `bookmarkEvent` itself is `useCallback`ed, so it should be good

Reply:
Hm... that still makes new functions on each render, but then, if `bookmarkEvent` is alread memoized... Interesting to ponder it some other time. Looks OK either way